### PR TITLE
community: Update ChatZhipuAI to support GLM-4 model

### PR DESCRIPTION
**Pull request by zhangch9**
Description: Update `ChatZhipuAI` to support the latest `glm-4` model.
Issue: N/A
Dependencies: httpx, httpx-sse, PyJWT

The previous `ChatZhipuAI` implementation requires the `zhipuai` package, and cannot call the latest GLM model. This is because
- The old version `zhipuai==1.*` doesn't support the latest model.
- `zhipuai==2.*` requires `pydantic V2`, which is incompatible with 'langchain-community'.

This re-implementation invokes the GLM model by sending HTTP requests to [open.bigmodel.cn](https://open.bigmodel.cn/dev/api) via the `httpx` package, and uses the `httpx-sse` package to handle stream events.

### CODE REVIEW

**Comment by zRzRzRzRzRzRzR** [comment]
We are still modifying the code and may need to resolve dependency conflicts during the review process

**Comment by zRzRzRzRzRzRzR** [comment]
We have completed the modification and would like to know if it can be merged

**Comment by zhangch9** [comment]
@hwchase17 @baskaryan Can you help review this pr? Let me know if any changes are required, thanks.

**Comment by baskaryan** [comment]
cc @linancn @whucalrence @JaneCrystall

**Review by linancn — APPROVED** [review]
I have successfully tested this PR, **LGTM**.
The notebook needs further refinement in subsequent PRs.

**Review by baskaryan — APPROVED** [review]

**Comment by zRzRzRzRzRzRzR** [comment]
so I need to add a use case notebook and it will done? thk

**Comment by jianchuanqi** [comment]
Hi there. I've noticed that this PR has not had any updates for the past two weeks. Could you please provide an update on its status, or let us know if any further information or modifications are needed? Thank you for your time and assistance.

**Comment by jianchuanqi** [comment]
> Hi there. I've noticed that this PR has not had any updates for the past two weeks. Could you please provide an update on its status, or let us know if any further information or modifications are needed? Thank you for your time and assistance.

@baskaryan  @hwchase17 Hi guys, any update?

**Comment by zRzRzRzRzRzRzR** [comment]
any update?

**Comment by baskaryan** [comment]
@zhangch9 there have been some changes to ChatZhipuAI, is this pr still needed or has the issue been resolved?

**Comment by linancn** [comment]
@baskaryan @zhangch9
This PR is precisely for the key improvement to transition from ZhipuAI API V3 to the latest V4 compatibility, and it needs to be merged.

File: libs/community/langchain_community/chat_models/zhipuai.py [hunk 1]
```diff
@@ -49,198 +162,131 @@ class ChatZhipuAI(BaseChatModel):
...
-                prompt=prompt,
-                meta=_meta,
-                request_id=self.request_id,
-                return_type=self.return_type,
-                incremental=self.incremental,
-            )
-        return None
+    # @property
+    # def _identifying_params(self) -> Dict[str, Any]:
+    #     return {"model_name": self.model}
```
Comment (baskaryan) [inline libs/community/langchain_community/chat_models/zhipuai.py:252]: delete

**Review by baskaryan — COMMENTED** [review]
looks like there's some merge conflicts that need to be resolved, otherwise lgtm

**Comment by zhangch9** [comment]
@baskaryan Conflicts resolved. PTAL.

**Pull request merged**